src/bin/pg_autoctl/primary_standby.c

```diff
@@ -432,17 +439,27 @@ primary_create_replication_user(LocalPostgresServer *postgres, char *replication
  * to pg_hba.conf on the primary.
  */
 bool
-primary_add_standby_to_hba(LocalPostgresServer *postgres, char *standbyHostname,
+primary_add_standby_to_hba(LocalPostgresServer *postgres,
+						   char *standbyHostname,
 						   const char *replicationPassword)
 {
 	PGSQL *pgsql = &(postgres->sqlClient);
 	PostgresSetup *postgresSetup = &(postgres->postgresSetup);
 	char hbaFilePath[MAXPGPATH];
-	char *authMethod =  "trust";
+	char *authMethod = pg_setup_get_auth_method(postgresSetup);
 
-	if (replicationPassword)
+	if (replicationPassword == NULL)
 	{
-		authMethod = pg_setup_get_auth_method(postgresSetup);
+		/* most authentication methods require a password */
+		if (strcmp(authMethod, "trust") != 0
```

Comment:
consider using a macro like HBA_TRUST, so that we won't have hardcoded "trust" in the c code

Reply:
I don't think we would gain anything that way. Maybe we could have a quick parser for HBA authentication methods and use an enum internally? I didn't do that, as I'm not sure it's necessary to go that far at the moment.